Add accessibility documentation

HedwigAR opened the pull request:
Here's the first draft of new accessibility documentation. The goal of this is to explain accessibility in general and give some pointers for people how to make their apps more accessible. It's very much meant as a starting point that later documentation can build off of. 
It currently still needs some images of certain settings but the text should be all there already.

HedwigJDoets committed:
initial draft of accessibility how to

HedwigJDoets committed:
Add urls

HedwigJDoets committed:
Update page titled description

HedwigJDoets committed:
Move documentation from how-to to refguide

HedwigJDoets committed:
Update document after review

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/mendix/docs?pullRequest=10220) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/mendix/docs?pullRequest=10220) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: ConnorLand<br/>:x: HedwigJDoets<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/mendix/docs?pullRequest=10220) it.</sub>

ConnorLand committed:
Improve text

ConnorLand committed:
edit

ConnorLand committed:
Edit

ConnorLand committed:
edit

ConnorLand committed:
edit and add key styling

ConnorLand commented on content/en/docs/refguide/general/accessibility-best-practices.md:
```diff
@@ -0,0 +1,250 @@
+
+### Link Purpose
+
+[Link purpose](https://www.w3.org/TR/WCAG22/#link-purpose-in-context), instead of using **Click here**, the link's destination should be clear from the surrounding text. Ideally, that description would be part of the link itself. It would help users understand the purpose of each link, enabling them to decide whether to follow it. Also, assistive technology can provide users with a list of links  ...
```
What does this mean "instead of using **Click here**? I think this example needs to be a bit more spelled out.

HedwigAR replied:
This refers to a situation where you have a link in a block of text. For example, on a museum website you could have something like: `[Click here](url) to learn more about fossils`
This would not comply with accessibility best practice. Instead people should do something more like: `To learn more about fossils, go to our [fossil collection](url)`

I'll add a mendix specific example

ConnorLand committed:
Edit

ConnorLand committed:
Edit

ConnorLand committed:
Edit

HedwigJDoets committed:
Expand explanation of link purpose with examples

HedwigJDoets committed:
Add images to doc

ConnorLand committed:
Put alt text into image codes

ConnorLand commented:
Edits on this doc are mostly done. The work remaining is to:

1. Make a new parent doc in Modeling titled something related to best practices
2. Move this doc and 2 others into the new category
3. Fix broken links
4. Set up aliases in the moved docs' metadata

I will finish that work when I get back.

ConnorLand committed:
Edit

ConnorLand committed:
update title

ConnorLand committed:
move a11y best practices to modelling

making a new parent doc, then will put a117 doc, Community Best Practices for App Performance, and Developer Tool Recommendations under it

ConnorLand committed:
Add parent doc

ConnorLand committed:
move images

ConnorLand committed:
update image URLs

ConnorLand committed:
add code tics

ConnorLand committed:
Add list property to parent doc

ConnorLand committed:
move dev tools and app performance docs

ConnorLand committed:
change menu order

ConnorLand committed:
add underscore to filename

ConnorLand committed:
move dev best practices

ConnorLand committed:
fix links

ConnorLand committed:
test commit for travis

ConnorLand committed:
Improve apostrophes

ConnorLand commented:
Good to merge after Hedwig and I resolve conflicts.
